### PR TITLE
Support loading multiple yamls in a file

### PR DESCRIPTION
**Pull request by ksivamuthu**
# Description

Used `YAML.v2` library  that supports reading multi-resource YAML files

Support Multi resource YAML files. The standalone component loader loads the YAML from the `components` directory. If the YAML has multi resources separated by `---`, only the first component is loaded. The other components are skipped. 

Using the decoder method, the standalone component loader unmarshals all the resources in YAML.

For e.g

```
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: statestore
spec:
  type: state.couchbase
  metadata:
  - name: couchbaseURL
    value: http://localhost:8091
  - name: username
    value: "TESTUSER"
  - name: password
    value: "PASSWORD"
  - name: bucketName
    value: dapr-state
---
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: receive-event
spec:
  type: bindings.aws.kinesis
  metadata:
  - name: region
    value: us-east-1
  - name: accessKey
    value: AWS_ACCESS_KEY # replace
  - name: secretKey
    value: AWS_SECRET_KEY #replace
  - name: streamName
    value: STREAM_NAME # Kinesis stream name
  - name: mode
    value: shared # shared - Shared throughput or extended - Extended fanout
```

Previous:
```
INFO[0000] standalone mode configured                    app_id=node-subscriber instance=CEI-840.local scope=dapr.runtime type=log ver=edge
INFO[0000] dapr id: node-subscriber                      app_id=node-subscriber instance=CEI-840.local scope=dapr.runtime type=log ver=edge
INFO[0000] mTLS is disabled. Skipping certificate request and tls validation  app_id=node-subscriber instance=CEI-840.local scope=dapr.runtime type=log ver=edge
INFO[0000] found component statestore (state.couchbase)  app_id=node-subscriber instance=CEI-840.local scope=dapr.runtime type=log ver=edge
INFO[0000] application protocol: http. waiting on port 3000  app_id=node-subscriber instance=CEI-840.local scope=dapr.runtime type=log ver=edge
```
After Change:

```
INFO[0000] standalone mode configured                    app_id=node-subscriber instance=CEI-840.local scope=dapr.runtime type=log ver=edge
INFO[0000] dapr id: node-subscriber                      app_id=node-subscriber instance=CEI-840.local scope=dapr.runtime type=log ver=edge
INFO[0000] mTLS is disabled. Skipping certificate request and tls validation  app_id=node-subscriber instance=CEI-840.local scope=dapr.runtime type=log ver=edge
INFO[0000] found component  (bindings.aws.kinesis)       app_id=node-subscriber instance=CEI-840.local scope=dapr.runtime type=log ver=edge
INFO[0000] found component  (state.couchbase)            app_id=node-subscriber instance=CEI-840.local scope=dapr.runtime type=log ver=edge
INFO[0000] application protocol: http. waiting on port 3000  app_id=node-subscriber instance=CEI-840.local scope=dapr.runtime type=log ver=edge

```

## Issue reference

PR will close: #1298 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation
* [ ] Specification has been updated
* [x] Provided sample for the feature

### CODE REVIEW

**Comment by yaron2** [comment]
Notice that the Google yaml package now caused the name of the component to not be serialized..

You can see it in the log examples above.

File: pkg/components/standalone_loader.go [hunk 1]
```diff
@@ -40,18 +42,25 @@ func (s *StandaloneComponents) LoadComponents() ([]components_v1alpha1.Component
 		if !file.IsDir() {
```
Comment (youngbupark) [inline pkg/components/standalone_loader.go]: Can you please add new test for LoadComponents() as a simple go test ?

Comment (amanbha) [reply pkg/components/standalone_loader.go]: +1 on new test.

Comment (ksivamuthu) [reply pkg/components/standalone_loader.go]: Sure. I will add.

**Comment by ksivamuthu** [comment]
> Notice that the yaml.v2 package now caused the name of the component to not be deserialized.
> 
> You can see it in the log examples above.

The YAML v2 is NOT accepting `json` tag in a struct.

Couple of approaches we can take here.

1. Add `yaml` tags along with `json` tag in the components struct. 
2. Split the YAML file into multiple resources using the separator and process them individually using the `ghodss` library.

**Comment by ksivamuthu** [comment]
After looking into the metadata, I see it includes metadata from k8s.io/apimachinery/pkg/apis/meta/v1, which is json tagged. In this case, I'll drop using gopkg.in/yaml.v2 and keep the previous `ghodss` library for `json` tag support. 

Instead, I will add the split YAML method using separator and pass the parsed individual yaml file to unmarshal.

File: pkg/components/standalone_loader.go (hunk 1 above)
Comment (ksivamuthu) [reply pkg/components/standalone_loader.go]: Added unit tests for decoding yamls.

File: pkg/components/standalone_loader.go [hunk 2]
```diff
@@ -37,23 +44,93 @@ func (s *StandaloneComponents) LoadComponents() ([]components_v1alpha1.Component
...
+			components, _ := s.DecodeYaml(fmt.Sprintf("%s/%s", dir, file.Name()), b)
+			list = append(list, components...)
 		}
 	}
 
 	return list, nil
 }
+
+// IsYaml checks whether the file is yaml or not
+func (s *StandaloneComponents) IsYaml(fileName string) bool {
```
Comment (youngbupark) [inline pkg/components/standalone_loader.go]: Can you please lowercase new method names?  because these are used only in this file

Comment (ksivamuthu) [reply pkg/components/standalone_loader.go]: Pretty new to "Go". Thanks for the comment. Done the changes

File: pkg/components/standalone_loader.go [hunk 3]
```diff
@@ -37,23 +44,93 @@ func (s *StandaloneComponents) LoadComponents() ([]components_v1alpha1.Component
...
-			list = append(list, component)
+			components, _ := s.decodeYaml(fmt.Sprintf("%s/%s", dir, file.Name()), b)
+			list = append(list, components...)
 		}
 	}
 
 	return list, nil
 }
+
+// IsYaml checks whether the file is yaml or not
```
Comment (youngbupark) [inline pkg/components/standalone_loader.go]: nit:  please lowercase method names in comments :)

**Comment by youngbupark** [comment]
@amanbha @yaron2 Can you please take a look at this pr again ?

**Pull request merged**